### PR TITLE
Require Node.js 12

### PR DESCRIPTION
**Pull request by sindresorhus**
Node.js 10 is soon out of LTS (in April), so I think it makes sense to do it for this major release.

This PR also prepares us for ESM. The ESM move is blocked by: https://github.com/avajs/typescript/issues/5#issuecomment-754678634

### CODE REVIEW

**Comment by sindresorhus** [comment]
@szmarczak I tried to remove https://github.com/sindresorhus/got/blob/23695808da78b689b79febb9d5ff3a7061bf578d/source/core/index.ts#L1391-L1392 and it's relevant code, but tests started failing. How do you intend to remove it?

File: benchmark/index.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
-'use strict';
 import {URL} from 'url';
-import https = require('https');
+import * as https from 'https';
```
Comment (szmarczak) [inline benchmark/index.ts:2]: Why this change?

Comment (sindresorhus) [reply benchmark/index.ts:2]: To prepare for ESM. `= require(…)` is not allowed when compiling to native ESM.

**Comment by szmarczak** [comment]
> it's relevant code, but tests started failing. How do you intend to remove it?

Let me see...

**Comment by szmarczak** [comment]
`pem` crashes on `macOS`

**Comment by szmarczak** [comment]
I'm not really sure why but this line fails on Node.js 12 only: https://github.com/nodejs/node/blob/v12.20.1/lib/_http_client.js#L472 only when running timeout tests. If a timeout gets breached then Node.js throws that assertion error.

**Comment by sindresorhus** [comment]
> pem crashes on macOS

That's an issue in `master` too.

> I'm not really sure why but this line fails on Node.js 12 only: https://github.com/nodejs/node/blob/v12.20.1/lib/_http_client.js#L472 only when running timeout tests. If a timeout gets breached then Node.js throws that assertion error.

This is also an existing issue in `master`. Should definitely be fixed, but not really relevant to this PR.

**Review by szmarczak — APPROVED** [review]
